File: src/Frontend/React/TypeScript/App/src/domains/Topic/Operations/Item/Get/Request/TopicDomainItemGetOperationRequest.ts
import {
  type ApiRequestWithInput,
  type TopicDomainItemGetOperationInput
} from '../../../../../../all';

export interface TopicDomainItemGetOperationRequest
  extends ApiRequestWithInput<TopicDomainItemGetOperationInput> {}

export function createTopicDomainItemGetOperationRequest (
  input: TopicDomainItemGetOperationInput,
  operationCode: string = ''
): TopicDomainItemGetOperationRequest {
  return {
    operationName: '@@TopicDomainItemGet',
    operationCode,
    input
  };
}

File: src/Backend/Services/FileStorage/Domains.Topic/Operations/Item/Get/TopicDomainItemGetOperationRequest.cs
// Copyright (c) 2023 Maxim Kuzmin. All rights reserved. Licensed under the MIT License.

namespace Crib2023.Backend.Services.FileStorage.Domains.Topic.Operations.Item.Get;

/// <summary>
/// Запрос операции получения элемента в домене.
/// </summary>
public class TopicDomainItemGetOperationRequest : IRequest<TopicDomainItemGetOperationResponse>
{
    #region Properties

    /// <summary>
    /// Входные данные.
    /// </summary>
    public TopicDomainItemGetOperationInput Input { get; }

    /// <summary>
    /// Код операции.
    /// </summary>
    public string OperationCode { get; }

    #endregion Properties

    #region Constructors

    /// <summary>
    /// Конструктор.
    /// </summary>
    /// <param name="input">Входные данные.</param>
    /// <param name="operationCode">Код операции.</param>
    public TopicDomainItemGetOperationRequest(TopicDomainItemGetOperationInput input, string operationCode = "")
    {
        Input = input;
        OperationCode = operationCode;
    }

    #endregion Constructors
}
